regions_www/m/server.R

```diff
@@ -19,804 +19,1071 @@
+      )
+      local_line_types <- c("None"                   = "none",
+                            "Route Network (LSOA)"   = "route_network"
+      )
+      local_line_order <- c(
+        "Number of cycle trips" = "slc",
+        "Increase in cyclists"   = "sic"
+      )
+
+    } else if (purpose == "alltrips") {
+      local_scenarios <- c(
+        "Current travel patterns" = "olc",
+        "Government Target"       = "govtarget",
+        "Gender equality"         = "gendereq",
+        "Go Dutch"                = "dutch",
+        "Ebikes"                  = "ebike"
+      )
+      local_line_types <- c("None"                   = "none",
+                            "Straight Lines"         = "straight_lines",
+                            "Fast Routes"            = "routes_fast",
+                            "Fast & Quieter Routes"  = "routes",
+                            "Route Network (MSOA)"   = "route_network"
+      )
+      local_line_order <- c(
+        "Number of cycle trips" = "slc",
+        "Increase in Cycling"   = "sic",
+        "HEAT Value"            = "slvalue_heat",
+        "CO2 reduction"         = "sico2"
+      )
+
+    }
+
+    # Update line options available, and set to default if the currently-selected option does not exist
+    if (input$scenario %in% local_scenarios) {
+      updateSelectInput(session, "scenario", choices = local_scenarios, selected = input$scenario)
+    } else {
+      updateSelectInput(session, "scenario", choices = local_scenarios)
+    }
+
+    if (input$line_type %in% local_line_types) {
+      updateSelectInput(session, "line_type", choices = local_line_types, selected = input$line_type)
+    } else {
+      updateSelectInput(session, "line_type", choices = local_line_types)
+    }
+
+    if (input$line_order %in% local_line_order) {
+      updateSelectInput(session, "line_order", choices = local_line_order, selected = input$line_order)
+    } else {
+      updateSelectInput(session, "line_order", choices = local_line_order)
+    }
+
+  }
+
+  ##############
+  # Initialise region and to_plot, and update right hand menu by region/purpose
+  ##############
+
+  ## Create region, to_plot and (for persistent geographical values) helper
+  region <- reactiveValues(current = NA, data_dir = NA, geography = NA, repopulate_region = F, purposes_present = NA)
+  to_plot <- NULL
+  helper <- NULL
+  helper$e_lat_lng <- ""
+
+  ## Set  values of region
   observe({
-    query <- parseQueryString(session$clientData$url_search)
+    region$current
+    region$data_dir
+    region$geography
+    region$repopulate_region
+    region$purposes_present
+    input$purpose
 
+    # Identify region
+    query <- parseQueryString(session$clientData$url_search)
     if (is.na(region$current)) {
-      region$current <- if(isTRUE(query[['r']] %in% regions$Region)){
+      region$current <- if (isTRUE(query[['r']] %in% regions$region_name)) {
         query[['r']]
-      } else if (exists("starting_region")) {
```

Comment:
This was to allow us to set the starting region without editing the code, i.e. in a local R session

```R
starting_region <- "lincolnshire"
```

Fine to remove but it has been a useful feature.

Reply:
It's still useful and still works for local testing, as you'll see by running it:

```r
shiny::runApp('regions_www/m')
```

@AnnaGoodman1 set the default to the Isle of Wight which is even more useful than Lincolnshire or (I'm sad to say) West Yorkshire.

Reply:
@Robinlovelace 
 
```R
starting_region <- "lincolnshire" # Or anywhere!
shiny::runApp('regions_www/m')
```
Used to start the app in Lincolnshire (or anywhere else we picked).

Reply:
I see,  I think I couldn't work out when/how it would be used hence deleting it –  if you find it useful, of course it can be added back